utility/unit_of_work/module.go

```diff
@@ -216,3 +113,46 @@ func (uow *baseUtilityUnitOfWork) Release() error {
 
 	return nil
 }
+
+func (uow *baseUtilityUnitOfWork) isProposalBlockSet() bool {
+	return uow.proposalStateHash != "" && uow.proposalProposerAddr != nil
```

Comment:
And check that `uow.proposalBlockTxs` is not empty

Reply:
I did it before but tests were failing so I assumed it was a valid case

Reply:
I did it before but tests were failing so I assumed it was a valid case to have empty TXs or maybe I didn't want to address the test(s) :P

Adding a TODO